### PR TITLE
Bug 1787745: Remove the RSA's md5 algorithm from the Ansible openssl modules.

### PR DESCRIPTION
**Pull request by timflannagan1**
When attempting to run Metering on a FIPS-enabled cluster, we fail role execution while generating user TLS certificates calling the 'openssl_privatekey' module. This module utilizes the module_utils/crypto.py helper library. This library iterates over a list of hashlib algorithms, which contains "FIPS-compliant" algorithms but still contains the 'md5' algorithm.

This would remove that algorithm from the list of available algorithms, and allow role execution to progress past the TLS-related tasks.

### CODE REVIEW

**Comment by timflannagan1** [comment]
CC @pruan-rht

File: images/metering-ansible-operator/roles/meteringconfig/module_utils/crypto.py [hunk 1]
```diff
@@ -0,0 +1,2015 @@
...
+def get_fingerprint_of_bytes(source):
+    """Generate the fingerprint of the given bytes."""
+
+    fingerprint = {}
+
+    try:
+        algorithms = hashlib.algorithms
+    except AttributeError:
+        try:
+            algorithms = hashlib.algorithms_guaranteed
```
Comment (pruan-rht) [inline images/metering-ansible-operator/roles/meteringconfig/module_utils/crypto.py]: @timflannagan1 the #guraanteed method returns a set, so I think you can just do algorithms.remove('md5') w/o transforming it into a `list`

Comment (timflannagan1) [reply images/metering-ansible-operator/roles/meteringconfig/module_utils/crypto.py]: Weirdly enough, when I import the hashlib in the python interpreter locally, I see the same thing:

```bash
tflannag@localhost operator-metering [docker-src-updates] ▶ python
Python 3.7.5 (default, Oct 17 2019, 12:16:48) 
[GCC 9.2.1 20190827 (Red Hat 9.2.1-1)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> 
>>> import hashlib
>>> 
>>> type(hashlib.algorithms_guaranteed)
<class 'set'>
>>> 
```

But when I was running this through the metering-operator, that method was being interpreted as a tuple, and not a set/list.

Comment (timflannagan1) [reply images/metering-ansible-operator/roles/meteringconfig/module_utils/crypto.py]: Resolved through slack - `hashlib.algorithms` returns a tuple, and `hashlib.algorithms_guaranteed` returns a set. We were assuming that `hashlib.algorithms_guaranteed` was always going to be returned as `hashlib.algorithms` was depreciated in python3. 

Metering uses a python2 build, so there's a fair chance that `hashlib.algorithms` will be available, and in the case it's not, `hashlib.algorithms_guaranteed` will be chosen. In either case, both get translated to a list so we should have consistent behavior.

File: images/metering-ansible-operator/roles/meteringconfig/module_utils/crypto.py [hunk 2]
```diff
@@ -0,0 +1,2015 @@
...
+    digest = None
+    if digest_string == 'sha256':
+        digest = hashes.SHA256()
+    elif digest_string == 'sha384':
+        digest = hashes.SHA384()
+    elif digest_string == 'sha512':
+        digest = hashes.SHA512()
+    elif digest_string == 'sha1':
+        digest = hashes.SHA1()
+    elif digest_string == 'md5':
```
Comment (EmilyM1) [inline images/metering-ansible-operator/roles/meteringconfig/module_utils/crypto.py]: Does this need to be removed?

File: images/metering-ansible-operator/roles/meteringconfig/module_utils/crypto.py [hunk 3]
```diff
@@ -0,0 +1,2015 @@
...
+    '1.2.840.113533.7.66.12': ('pbeWithMD5AndCast5CBC', ),
+    '1.2.840.113533.7.66.13': ('password based MAC', 'id-PasswordBasedMAC'),
+    '1.2.840.113533.7.66.30': ('Diffie-Hellman based MAC', 'id-DHBasedMac'),
+    '1.2.840.113549': ('RSA Data Security, Inc.', 'rsadsi'),
+    '1.2.840.113549.1': ('RSA Data Security, Inc. PKCS', 'pkcs'),
+    '1.2.840.113549.1.1': ('pkcs1', ),
+    '1.2.840.113549.1.1.1': ('rsaEncryption', ),
+    '1.2.840.113549.1.1.2': ('md2WithRSAEncryption', 'RSA-MD2'),
+    '1.2.840.113549.1.1.3': ('md4WithRSAEncryption', 'RSA-MD4'),
+    '1.2.840.113549.1.1.4': ('md5WithRSAEncryption', 'RSA-MD5'),
```
Comment (EmilyM1) [inline images/metering-ansible-operator/roles/meteringconfig/module_utils/crypto.py]: Unneeded?

File: images/metering-ansible-operator/roles/meteringconfig/module_utils/crypto.py (hunk 2 above)
Comment (timflannagan1) [reply images/metering-ansible-operator/roles/meteringconfig/module_utils/crypto.py]: Nope, the openssl Ansible modules don't interact with this function.

**Pull request merged**